File: solutions/go/bird-watcher/2/bird_watcher.go
package birdwatcher

func TotalBirdCount(birdsPerDay []int) int {
    birdCount := 0
    for _, birds := range birdsPerDay{
        birdCount += birds
    }
	return birdCount
}

func BirdsInWeek(birdsPerDay []int, week int) int {
	startOfWeek := 7  * (week - 1)
    stopOfWeek := startOfWeek + 7
    return TotalBirdCount(birdsPerDay[startOfWeek:stopOfWeek])
}

func FixBirdCountLog(birdsPerDay []int) []int {
	for i := 0; i < len(birdsPerDay); i += 2{
        birdsPerDay[i]++ 
    }
	return birdsPerDay
}

File: solutions/javascript/bird-watcher/2/bird-watcher.js
export function totalBirdCount(birdsPerDay) {
  let count = 0;
  for (let i=0; i < birdsPerDay.length; i++){
    count += birdsPerDay[i];
  }
  return  count;
}

export function birdsInWeek(birdsPerDay, week) {
  let startDay = (week - 1) * 7 ;
  return totalBirdCount(birdsPerDay.slice(startDay, startDay + 7));
}

export function fixBirdCountLog(birdsPerDay) {
  for (let i=0; i < birdsPerDay.length; i += 2) birdsPerDay[i]++ ;
  return birdsPerDay;
}
